beacon-chain/blockchain/error.go

```diff
@@ -17,3 +17,29 @@ var (
 	// block is not a valid optimistic candidate block
 	errNotOptimisticCandidate = errors.New("block is not suitable for optimistic sync")
 )
+
+type invalidBlock struct {
```

Comment:
We need to add extensive commentary here about what this is used for and what is considered valid vs. invalid. How should callers be using this and when should they be wrapping their errors this way? Should we be exporting this?